airflow/models/dagrun.py

```diff
@@ -124,7 +124,9 @@ class DagRun(Base, LoggingMixin):
         ),
     )
 
-    task_instances = relationship(TI, back_populates="dag_run")
+    task_instances = relationship(
+        TI, back_populates="dag_run", cascade='save-update, merge, delete, delete-orphan'
```

Comment:
```suggestion
        TI, back_populates="dag_run", cascade='all, delete-orphan'
```
I think? That seems to be what the docs recommend as "typical"

Reply:
Was confused about the many options. :)

Reply:
See https://github.com/apache/airflow/pull/18434#discussion_r713958136 (not sure, but the `explunge` part feels undesired)

Reply:
Hmm. `all` is short for `save-update, merge, refresh-expire, expunge, delete`

Reply:
expunging the deleted rows from the session is fine -- it's been deleted from the DB anyway.